Engine/Module/ModuleResourceManager.cpp

```diff
@@ -39,136 +79,244 @@ update_status ModuleResourceManager::PreUpdate()
+	thread_comunication.main_importing_hash = 0;
+	return imported_resource_uuid;
+}
+
+uint32_t ModuleResourceManager::InternalImport(Path& file_path) const
+{
+	//std::lock_guard<std::mutex> lock(thread_comunication.thread_mutex);
+
+	Metafile* asset_metafile = nullptr;
+
+	if (Importer::ImportRequired(file_path))
 	{
-		result = prefab_importer->Import(file, force);
+		switch (file_path.GetFile()->GetFileType())
+		{
+		case FileType::ANIMATION:
+			asset_metafile = animation_importer->Import(file_path);
+			break;
+		
+		case FileType::MATERIAL:
+			asset_metafile = material_importer->Import(file_path);
+			break;
+
+		case FileType::MESH:
+			asset_metafile = mesh_importer->Import(file_path);
+			break;
+
+		case FileType::MODEL:
+			asset_metafile = model_importer->Import(file_path); // WARNING! If you load a model dont forget to put Prefab overwritable flag to false!!
+			break;
+
+		case FileType::PREFAB:
+			asset_metafile = prefab_importer->Import(file_path);
+			break;
+
+		case FileType::SKELETON:
+			asset_metafile = skeleton_importer->Import(file_path);
+			break;
+
+		case FileType::TEXTURE:
+			asset_metafile = texture_importer->Import(file_path);
+			break;
+
+		case FileType::SKYBOX:
+			asset_metafile = skybox_importer->Import(file_path);
+			break;
+		}
 	}
-	if (file.file_type == FileType::MESH)
+	else
 	{
-		result = model_importer->ImportExtractedResources(file, force);
+		Path* metafile_path = App->filesystem->GetPath(metafile_manager->GetMetafilePath(file_path));
+		asset_metafile = metafile_manager->GetMetafile(*metafile_path);
 	}
-	return result;
+
+	resource_DB->AddEntry(asset_metafile);
+
+	return asset_metafile->uuid;
 }
 
-std::shared_ptr<Resource> ModuleResourceManager::RetrieveFromCacheIfExist(const std::string& uid) const
+std::shared_ptr<Resource> ModuleResourceManager::Load(uint32_t uuid)
 {
-	if (!App->filesystem->Exists(uid.c_str()))
+	BROFILER_CATEGORY("Load Resource", Profiler::Color::Brown);
+	APP_LOG_INFO("Loading Resource %u.", uuid);
+
+	std::shared_ptr<Resource> loaded_resource;
+	loaded_resource = RetrieveFromCacheIfExist(uuid);
+	if (loaded_resource != nullptr)
 	{
+		APP_LOG_SUCCESS("Resource %u loaded correctly from cache.", uuid);
+		return loaded_resource;
+	}
+
+	Metafile* metafile = resource_DB->GetEntry(uuid);
+	if (!App->filesystem->Exists(metafile->exported_file_path))
+	{
+		APP_LOG_ERROR("Error loading Resource %u. File %s doesn't exist", uuid, metafile->exported_file_path);
 		return nullptr;
 	}
-	//Check if the resource is already loaded
-	auto it = std::find_if(resource_cache.begin(), resource_cache.end(), [uid](const std::shared_ptr<Resource> resource)
+
+	Path* resource_exported_file_path = App->filesystem->GetPath(metafile->exported_file_path);
+	FileData exported_file_data = resource_exported_file_path->GetFile()->Load();
+
+	switch (metafile->resource_type)
 	{
-		return resource->exported_file == uid;
-	});
-	if (it != resource_cache.end())
+	case ResourceType::ANIMATION:
```

Comment:
You now I created the Load template exactly in order to avoid this... I do understand that maybe is cooler to load something by just calling a method and don't knowing the type. But I hate having switch and having switches everywere usually is a bad praxis.

Reply:
I do agree, I will change it